### PR TITLE
importccl: allow literal values in mysqldump DEFAULTs

### LINKED ISSUE #29301: importccl: IMPORT MYSQLDUMP says "variable sub-expressions are not allowed in DEFAULT"
Testing with [this MySQL dump file](https://raw.githubusercontent.com/artmarkov/smart.loc/50b43d5640dcfcc38f2c7db22f04982e43b93a09/backup/db/db__2017-11-19_16-38-09.sql) I found on GitHub, CockroachDB gave me this error:

```
root@:26257/defaultdb> import mysqldump ('https://raw.githubusercontent.com/artmarkov/smart.loc/50b43d5640dcfcc38f2c7db22f04982e43b93a09/backup/db/db__2017-11-19_16-38-09.sql');
pq: variable sub-expressions are not allowed in DEFAULT
```

I'm kind of lost on this one, because I looked through the dump file and didn't find any DEFAULT expressions that looked suspicious at first glance.

### LINKED ISSUE #29309: importccl: IMPORT MYSQLDUMP fails with "unsupported default expression"
Testing with [this MySQL dump file](https://raw.githubusercontent.com/cchou-git/DTCC_JavaAcademy/5cda05fd8001e0e0224c043078f572c4ffe3f2dd/appDataModel/src/database/sql/dtcc_db_script.sql) I found on GitHub, CockroachDB gave me this error:

```
root@:26257/defaultdb> import mysqldump ('https://raw.githubusercontent.com/cchou-git/DTCC_JavaAcademy/5cda05fd8001e0e0224c043078f572c4ffe3f2dd/appDataModel/src/database/sql/dtcc_db_script.sql');
pq: unsupported default expression for column "address1": syntax error at or near ")"
```

There are two table definitions with an "address1" column in this dump, both of which are defined like this example:

```sql
-- TABLE STRUCTURE FOR TABLE `RS_CLIENT`
--

DROP TABLE IF EXISTS `RS_CLIENT`;
/*!40101 SET @SAVED_CS_CLIENT     = @@CHARACTER_SET_CLIENT */;
/*!40101 SET CHARACTER_SET_CLIENT = UTF8 */;
CREATE TABLE `RS_CLIENT` (
  `NAME` VARCHAR(32) NOT NULL,
  `DISPLAY_NAME` VARCHAR(256) NOT NULL,
  `IS_DELETED` VARCHAR(1) DEFAULT 'N',
  `ADDRESS1` VARCHAR(256) DEFAULT '',
  `ADDRESS2` VARCHAR(256) DEFAULT '',
  `CITY` VARCHAR(256) DEFAULT '',
  `STATE` VARCHAR(256) DEFAULT '',
  `ZIP` VARCHAR(32) DEFAULT '',
  `COUNTRY` VARCHAR(256) DEFAULT '',
  `ID` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT,
  `VERSION` INT(10) UNSIGNED DEFAULT NULL,
  `SHOW_KEYFIND_NOTES` VARCHAR(1) DEFAULT 'N',
  `IS_LOGOUT_BTN_SHOWN` VARCHAR(1) DEFAULT 'Y',
  PRIMARY KEY (`ID`),
  UNIQUE KEY `ID` (`ID`),
  UNIQUE KEY `CLIENT_NAME` (`NAME`)
) ENGINE=INNODB AUTO_INCREMENT=6 DEFAULT CHARSET=UTF8;
/*!40101 SET CHARACTER_SET_CLIENT = @SAVED_CS_CLIENT */;
```

### PR DESCRIPTION
**Pull request by dt**
Previously we were failing to type literal strings as string expressions, instead thinking they were func or vars.
Wrap mysql's StrVals in quotes before handing them to the parser to have them correctly handled as literals.

Release note: none.

Fixes #29301.
Fixes #29309.

### CODE REVIEW

File: pkg/ccl/importccl/read_import_mysql.go [hunk 1]
```diff
@@ -598,13 +598,16 @@ func mysqlColToCockroach(
 	}
 
 	if col.Default != nil && !bytes.EqualFold(col.Default.Val, []byte("null")) {
-		expr, err := parser.ParseExpr(string(col.Default.Val))
+		exprString := string(col.Default.Val)
+		if col.Default.Type == mysql.StrVal {
+			exprString = fmt.Sprintf("'%s'", col.Default.Val)
```
Comment (mjibson) [inline pkg/ccl/importccl/read_import_mysql.go]: could you add a test for a default containing a single quote to make sure escaping happens properly?

Comment (dt) [reply pkg/ccl/importccl/read_import_mysql.go]: Done. Decided I'd rather skip the parser than try to quote/escape just to make it though the parser.

**Review by madelynnblue — APPROVED** [review]

**Comment by dt** [comment]
bors r+

this time hopefully examples ORMs doesn't flake it

**Pull request merged**